src/NBitcoin/Network.cs

```diff
@@ -10,1429 +12,1079 @@
+            {
+                return this.minimumChainWork;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                this.minimumChainWork = value;
+            }
+        }
+
+        public long DifficultyAdjustmentInterval
+        {
+            get
+            {
+                return ((long)PowTargetTimespan.TotalSeconds / (long)PowTargetSpacing.TotalSeconds);
+            }
+        }
+
+        int minerConfirmationWindow;
+        public int MinerConfirmationWindow
+        {
+            get
+            {
+                return this.minerConfirmationWindow;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                this.minerConfirmationWindow = value;
+            }
+        }
+
+        int ruleChangeActivationThreshold;
+        public int RuleChangeActivationThreshold
+        {
+            get
+            {
+                return this.ruleChangeActivationThreshold;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                this.ruleChangeActivationThreshold = value;
+            }
+        }
+
+        int coinType;
+
+        /// <summary>
+        /// Specify the BIP44 coin type for this network
+        /// </summary>
+        public int CoinType
+        {
+            get
+            {
+                return this.coinType;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                this.coinType = value;
+            }
+        }
+
+
+        bool litecoinWorkCalculation;
+        /// <summary>
+        /// Specify using litecoin calculation for difficulty
+        /// </summary>
+        public bool LitecoinWorkCalculation
+        {
+            get
+            {
+                return this.litecoinWorkCalculation;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                this.litecoinWorkCalculation = value;
+            }
+        }
+
+        BigInteger proofOfStakeLimit;
+        public BigInteger ProofOfStakeLimit
+        {
+            get
+            {
+                return proofOfStakeLimit;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                proofOfStakeLimit = value;
+            }
+        }
+
+        BigInteger proofOfStakeLimitV2;
+        public BigInteger ProofOfStakeLimitV2
+        {
+            get
+            {
+                return proofOfStakeLimitV2;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                proofOfStakeLimitV2 = value;
+            }
+        }
+
+        int lastPOWBlock;
+        public int LastPOWBlock
+        {
+            get
+            {
+                return lastPOWBlock;
+            }
+            set
+            {
+                EnsureNotFrozen();
+                lastPOWBlock = value;
+            }
+        }
+
+        private bool frozen = false;
```

Comment:
Do we need to set this to false?

Reply:
No but I am actually tempted to remove that logic all together.

Reply:
do it! suddenly you would get rid of the fields, would be much better

Reply:
ok I will